backend/dataall/modules/dataset_sharing/services/data_sharing_service.py

```diff
@@ -554,6 +585,17 @@ def reapply_share(cls, engine: Engine, share_uri: str):
                         )
                     return False
 
+            log.info(f'Verifying principal IAM Role {share.principalIAMRoleName}')
+            # If principal role doesn't exist, all share items are unhealthy, no use of further checks
+            if not ShareObjectService.verify_principal_role(session, share):
+                ShareObjectService.update_all_share_items_status(
+                    session,
+                    share_uri,
+                    ShareItemHealthStatus.Unhealthy,
+                    f'Share principal Role {share.principalIAMRoleName} is not found.',
+                )
+                return False
+
```

Comment:
We cannot update all share items `HealthStatus`  to `Unhealthy`, we can only update those whose `Status` is in `Share_Succeed`. Those items in other sharing states (e.g. `PendingApproval` do not have a share to be healthy or unhealthy)